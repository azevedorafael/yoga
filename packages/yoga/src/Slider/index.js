import Slider from './web/Slider';

export default Slider;
