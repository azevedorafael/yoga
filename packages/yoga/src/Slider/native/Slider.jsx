import React from 'react';
import { View } from 'react-native';
import { bool, number, arrayOf, shape, string } from 'prop-types';
import { withTheme } from 'styled-components';
import MultiSlider from '@ptomasroos/react-native-multi-slider';
import { LabelView, LabelText } from './Label';
import Marker from './Marker';
import Step from './Step';

const SliderComponent = ({
  max,
  min,
  maxLabel,
  minLabel,
  snapped,
  values,
  tooltip,
  theme: {
    components: {
      slider: {
        track: {
          backgroundColor: {
            active: activeBackgroundColor,
            inactive: inactiveBackgroundColor,
          },
          border: { radius },
        },
      },
    },
  },
  ...props
}) => {
  const renderSnapDots = () => {
    const items = [];
    for (let i = min; i <= max; i++) {
      if (values.length > 1) {
        items.push(<Step key={i} active={values[0] < i && values[1] > i} />);
      } else {
        items.push(<Step key={i} active={values[0] > i} />);
      }
    }

    return items;
  };

  const renderTooltip = side => {
    const tooltipSide = side === 'left' ? 0 : 1;

    return tooltip.filter(item => {
      if (!item.step && item.step !== 0) {
        return item.visible;
      }

      return item.visible && values[tooltipSide] === item.step;
    })[0];
  };

  return (
    <>
      {snapped && (
        <View
          style={{
            flexDirection: 'row',
            alignItems: 'center',
            justifyContent: 'space-between',
            bottom: -30,
          }}
        >
          {renderSnapDots()}
        </View>
      )}
      <View>
        <MultiSlider
          values={values}
          trackStyle={{
            backgroundColor: inactiveBackgroundColor,
            height: 4,
            borderRadius: radius,
          }}
          selectedStyle={{ backgroundColor: activeBackgroundColor }}
          isMarkersSeparated
          min={min}
          max={max}
          snapped={snapped}
          customMarkerLeft={({ pressed }) => (
            <Marker tooltip={renderTooltip('left')} pressed={pressed} />
          )}
          customMarkerRight={({ pressed }) => (
            <Marker tooltip={renderTooltip('right')} pressed={pressed} />
          )}
          {...props}
        />
        {(minLabel || maxLabel) && (
          <LabelView>
            {minLabel && <LabelText placement="left">{minLabel}</LabelText>}
            {maxLabel && <LabelText placement="right">{maxLabel}</LabelText>}
          </LabelView>
        )}
      </View>
    </>
  );
};

SliderComponent.propTypes = {
  max: number,
  maxLabel: string,
  min: number,
  minLabel: string,
  snapped: bool,
  tooltip: arrayOf(
    shape({
      description: string,
      title: string,
      ribbon: string,
      visible: bool,
      step: number,
    }),
  ),
  values: arrayOf(number),
};

SliderComponent.defaultProps = {
  max: 10,
  maxLabel: undefined,
  min: 0,
  minLabel: undefined,
  snapped: false,
  tooltip: [],
  values: [0],
};

export default withTheme(SliderComponent);
