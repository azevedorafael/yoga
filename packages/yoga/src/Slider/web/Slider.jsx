import React from 'react';
import { bool, string, number, arrayOf, shape } from 'prop-types';
import styled, { withTheme } from 'styled-components';
import RCSlider from 'rc-slider/lib/Slider';
import RCRange from 'rc-slider/lib/Range';
import Marker from './Marker';

const StyledSlider = styled(RCSlider)`
  position: relative;
`;

const StyledRange = styled(RCRange)`
  position: relative;
`;

const Slider = ({
  snapped,
  tooltip,
  values,
  theme: {
    components: {
      slider: {
        track: {
          backgroundColor: {
            active: activeBackgroundColor,
            inactive: inactiveBackgroundColor,
          },
          border: { radius: trackRadius },
        },
        step: {
          backgroundColor: {
            active: stepActiveBackgroundColor,
            inactive: stepInactiveBackgroundColor,
          },
          border: { radius: stepRadius },
        },
      },
    },
  },
  ...props
}) => {
  const commonTrackStyle = {
    borderRadius: trackRadius,
    height: 4,
    position: 'absolute',
    width: '100%',
  };

  const trackStyle = {
    backgroundColor: activeBackgroundColor,
  };

  const railStyle = {
    backgroundColor: inactiveBackgroundColor,
  };

  const commonStepStyle = {
    borderRadius: stepRadius,
    marginLeft: -4,
    position: 'absolute',
    height: 10,
    top: -3,
    width: 10,
  };

  const activeStepStyle = {
    backgroundColor: stepActiveBackgroundColor,
  };

  const inactiveStepStyle = {
    backgroundColor: stepInactiveBackgroundColor,
  };

  return values.length > 1 ? (
    <StyledRange
      {...props}
      dots={snapped}
      value={values}
      trackStyle={[
        {
          ...commonTrackStyle,
          ...trackStyle,
        },
      ]}
      railStyle={{
        ...commonTrackStyle,
        ...railStyle,
      }}
      dotStyle={{
        ...commonStepStyle,
        ...inactiveStepStyle,
      }}
      activeDotStyle={{
        ...commonStepStyle,
        ...activeStepStyle,
      }}
      handle={rest => <Marker values={values} tooltip={tooltip} {...rest} />}
    />
  ) : (
    <StyledSlider
      {...props}
      prefixCls="yoga"
      dots={snapped}
      value={values[0]}
      trackStyle={{
        ...commonTrackStyle,
        ...trackStyle,
      }}
      railStyle={{
        ...commonTrackStyle,
        ...railStyle,
      }}
      dotStyle={{
        ...commonStepStyle,
        ...inactiveStepStyle,
      }}
      activeDotStyle={{
        ...commonStepStyle,
        ...activeStepStyle,
      }}
      handle={rest => <Marker values={values} tooltip={tooltip} {...rest} />}
    />
  );
};

Slider.propTypes = {
  snapped: bool,
  tooltip: arrayOf(
    shape({
      ribbon: string,
      title: string,
      description: string,
      visible: bool,
      step: number,
    }),
  ),
  values: arrayOf(number),
};

Slider.defaultProps = {
  snapped: false,
  tooltip: undefined,
  values: [0],
};

export default withTheme(Slider);
