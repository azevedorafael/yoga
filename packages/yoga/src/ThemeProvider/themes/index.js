import corp from './corp';
import endUser from './end-user';
import gyms from './gyms';

export { endUser as default, endUser, corp, gyms };
