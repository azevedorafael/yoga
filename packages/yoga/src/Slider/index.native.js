import Slider from './native/Slider';

export default Slider;
