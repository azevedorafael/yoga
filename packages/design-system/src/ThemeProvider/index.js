import ThemeProvider from './ThemeProvider';
import * as themes from './themes';

export { ThemeProvider as default, themes };
