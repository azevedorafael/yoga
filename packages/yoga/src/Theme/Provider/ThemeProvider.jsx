import React from 'react';
import { ThemeProvider as SCThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import * as tokens from '@gympass/yoga-tokens';
import * as themes from '../themes';

const getTheme = ({ theme, locale }) => {
  const token = tokens[locale] || tokens.default;
  const appTheme = themes[theme] || themes.default;

  return appTheme(token);
};

const ThemeProvider = ({ children, ...theme }) => (
  <SCThemeProvider theme={{ yoga: getTheme(theme) }}>
    {children}
  </SCThemeProvider>
);

ThemeProvider.propTypes = {
  theme: PropTypes.string,
  locale: PropTypes.string,
  children: PropTypes.node.isRequired,
};

ThemeProvider.defaultProps = {
  theme: 'endUser',
  locale: 'pt-BR',
};

export default ThemeProvider;
