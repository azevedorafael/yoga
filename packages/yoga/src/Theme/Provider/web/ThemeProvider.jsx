import React from 'react';
import { node } from 'prop-types';

import Provider from '../Provider';
import GlobalStyle from './GlobalStyle';

/** This component provides a theme to all React components underneath itself via the context API. */
const ThemeProvider = ({ children, ...rest }) => (
  <Provider {...rest}>
    <GlobalStyle />
    {children}
  </Provider>
);

ThemeProvider.propTypes = {
  children: node.isRequired,
};

export default ThemeProvider;
