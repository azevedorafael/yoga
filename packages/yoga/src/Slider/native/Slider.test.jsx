import React from 'react';
import { render, fireEvent, toJSON } from '@testing-library/react-native';
import ThemeProvider from '../../ThemeProvider';
import Slider from '../';

describe('<Slider />', () => {
  describe('Snapshots', () => {
    describe('Without props', () => {
      it('should match snapshot with default Slider', () => {
        const { container } = render(
          <ThemeProvider>
            <Slider />
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });
    });
  });
});
