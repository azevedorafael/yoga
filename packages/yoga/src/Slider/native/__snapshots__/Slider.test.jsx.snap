// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshots Without props should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "height": 50,
            "justifyContent": "center",
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "width": 280,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#A7A7A7",
                "borderRadius": 2,
                "height": 2,
              },
              Object {
                "backgroundColor": "#DBDBE8",
                "borderRadius": 9999,
                "height": 4,
              },
              Object {
                "backgroundColor": "#F46152",
              },
              Object {
                "width": 0,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#A7A7A7",
                "borderRadius": 2,
                "height": 2,
              },
              Object {
                "backgroundColor": "#DBDBE8",
                "borderRadius": 9999,
                "height": 4,
              },
              undefined,
              Object {
                "width": 280,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "height": 48,
                "justifyContent": "center",
                "position": "absolute",
                "width": 48,
              },
              Object {
                "left": -24,
                "top": -24,
              },
              undefined,
              false,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 0,
                },
              ]
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              pressed={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "#F46152",
                    "borderColor": "black",
                    "borderRadius": 9999,
                    "borderStyle": "solid",
                    "borderWidth": 0,
                    "display": "none",
                    "elevation": 4,
                    "height": 24,
                    "position": "relative",
                    "shadowColor": "rgba(0, 0, 0, 0.25)",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 6,
                    "width": 24,
                  },
                ]
              }
            />
            <TouchableOpacity
              activeOpacity={0.2}
              style={
                Array [
                  Object {
                    "backgroundColor": "#F46152",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 9999,
                    "borderStyle": "solid",
                    "borderWidth": 6,
                    "elevation": 4,
                    "height": 24,
                    "position": "relative",
                    "shadowColor": "rgba(0, 0, 0, 0.25)",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 6,
                    "width": 24,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
